src/fu-tool.c

```diff
@@ -131,6 +131,25 @@ fu_util_stop_daemon (GError **error)
 	return val != NULL;
 }
 
+static gboolean
+fu_util_save_current_state (FuUtilPrivate *priv, GError **error)
+{
+	g_autoptr(GPtrArray) devs = NULL;
+	g_autofree gchar *state = NULL;
+	g_autofree gchar *dirname = NULL;
+	g_autofree gchar *filename = NULL;
+
```

Comment:
I wonder if we should only write the file if an argument is passed, e.g. `--enable-json-state` or something?

Reply:
Sure that's fine by me.